### DISCUSSION TITLE
Can't modify an existent client: 404

### DISCUSSION THREAD

**Discussion by paolocerto** [fattureincloud/api#273] [Q&A] [answered]
Hello,

I'm working on the upgrade to API v.2 with the PHP SDK
At today we do only 2 operations: insert a new client and update an old client.

I managed the insert new.
I can't upgrade an old client

This is the error message I get
Client error: `PUT https://api-v2.fattureincloud.it/c/COMPANY_ID/entities/clients/CLIENT_ID` resulted in a `404 Not Found`

all other code is the same as Insert new

$modify_client_request = new \FattureInCloud\Model\ModifyClientRequest;
$modify_client_request->setData($item);
$result = $apiInstance->modifyClient($id_company, $id_client, $modify_client_request);

Thank you for your help

Regards,
Paolo

**Comment by valmoz** [comment, marked as answer]
Hi @paolocerto 
please, make sure that the client_id refers to an existing client. How did you obtain that id?

Thanks
Mauro

**Comment by paolocerto** [comment]
Hi @valmoz 
client_id is stored after insert
or I should get it again?

Regards,
Paolo

**Reply by emajo** [reply]
You can use the id saved after creating the client, but make sure that the client still exists (it could have been deleted), I suggest to make a list clients call to check if the ids you saved are correct and still exist.

Thanks,
Emanuele.

**Comment by paolocerto** [comment]
Hi @emajo 

thank you for your suggestion: is the path I'm following.

Regards,
Paolo

**Reply by valmoz** [reply]
Hi @paolocerto were you able to solve your problem?

Thanks
Mauro

**Reply by paolocerto** [reply]
Hi @valmoz 
yes, at the end I discovered that all old clients disappeared
I have just the last inserted clients.
I don't know why
But the update was impossible for this reason.
Thank you

Regards,
Paolo
